src/index.ts

```diff
@@ -1,5 +1,129 @@
+   *
+   * @default InstanceType.of(InstanceClass.R5,InstanceSize.LARGE)
+   */
+  readonly instanceType?: InstanceType;
+  /**
+   * Kms Key - you probably want to provide one, but...
+   *
+   * @default - generate a KMS Key
+   */
+  readonly kmsKey?: IKey;
+  /**
+   * Removal Policy - applies to both the actual documentdb and the kms key (if created)
+   *
+   * @default - passthrough to RETAIN.
+   */
+  readonly removalPolicy?: RemovalPolicy;
+  /**
+   * Security group.
+   *
+   * @default undefined a new security group is created.
+   */
+  readonly securityGroup?: ISecurityGroup;
+  /**
+   * Which VPC?
+   */
+  readonly vpc: IVpc;
+}
+
+export class DocumentDb extends Construct {
+  public readonly cluster: DatabaseCluster;
+  public readonly kmsKey: IKey;
+  public readonly secret: DatabaseSecret;
+  public readonly securityGroup: ISecurityGroup;
+
+  constructor(scope: Construct, id: Namer, props: DocumentDbProps) {
+    super(scope, id.pascal);
+
+    const dbClusterName = props.dbClusterName ?? id.pascal;
+    const deletionProtection = props.deletionProtection ?? true;
+    const instances = props.instances ?? 1;
+    const instanceType = props.instanceType ?? InstanceType.of(InstanceClass.R6G, InstanceSize.LARGE);
+
+    this.kmsKey =
+      props.kmsKey ??
+      new Key(this, 'KmsKey', {
+        description: `aws_docdb ${dbClusterName}`,
+        removalPolicy: props.removalPolicy,
+      });
+
+    const parameterGroupName = id.addSuffix(['parameter', 'group']);
+    const parameterGroup = new ClusterParameterGroup(this, parameterGroupName.pascal, {
+      dbClusterParameterGroupName: parameterGroupName.pascal,
+      family: 'docdb4.0', // must match the DatabaseCluster below (we are using implicit)
+      parameters: {
+        audit_logs: 'enabled', // TODO: "enabled"? "true"? or... ???
+      },
+    });
+
+    this.cluster = new DatabaseCluster(this, 'Database', {
+      ...props,
+      masterUser: {
+        username: 'admin',
+        excludeCharacters: ' ;+%{}' + '@\'"`/\\#', // DMS and BASH problem chars. Keep life as easy as possible for aws_dms.
+        secretName: id.addSuffix(['admin']).pascal,
+      },
+      backup: props.backup ?? { retention: Duration.days(30) },
+      dbClusterName,
+      deletionProtection,
+      // engineVersion: '4.0', // while it would be nice to explicitly match this to the ParameterGroup above, it forces a replace of the instances. Implicit looks like a better pattern here.
+      instances,
+      instanceType,
+      kmsKey: this.kmsKey,
+      parameterGroup,
+      removalPolicy: props.removalPolicy,
+      securityGroup: props.securityGroup,
+      vpc: props.vpc,
+      vpcSubnets: { subnetType: SubnetType.PRIVATE_WITH_NAT }, // AWS does not allow putting DocDBs in public subnets.
+    });
+    this.securityGroup =
+      props.securityGroup || SecurityGroup.fromSecurityGroupId(this, 'SecurityGroup', this.cluster.securityGroupId);
```

Comment:
If the securityGroup is provided, it needs to have the appropriate ingress rules already set up, correct?  Anything this construct can do to validate and/or warn if it's not set up appropriately? (thinking probably not, at least not at synth time)

Reply:
If that mattered enough I'd argue the SG should just be created here. The flexibility to provide any SG with any set of ingress/egress rules is appealing.

Reply:
The docs just say:

```
securityGroup?
Type: ISecurityGroup (optional, default: a new security group is created.)

Security group.
```

So... I'm assuming that if we provide the SG, then the DocDB gets created in that SG.
If we don't, the construct creates an SG to host the DocDB.

Access to this SG would be granted elsewhere. And... what this really tells me is that we need to provide a README.md.

Reply:
Ah, yeah looks like they have a separate set of methods via the `connections` property for handling this ([ref](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_docdb-readme.html#connecting)).  Which is actually really nice, makes things more explicit.